parsl/executors/high_throughput/htex.py

```diff
@@ -0,0 +1,392 @@
+
+    def _start_queue_management_thread(self):
+        """Method to start the management thread as a daemon.
+
+        Checks if a thread already exists, then starts it.
+        Could be used later as a restart if the management thread dies.
+        """
+        if self._queue_management_thread is None:
+            logger.debug("Starting queue management thread")
+            self._queue_management_thread = threading.Thread(target=self._queue_management_worker)
+            self._queue_management_thread.daemon = True
+            self._queue_management_thread.start()
+            logger.debug("Started queue management thread")
+
+        else:
+            logger.debug("Management thread already exists, returning")
+
+    def submit(self, func, *args, **kwargs):
+        """Submits work to the the outgoing_q.
+
+        The outgoing_q is an external process listens on this
+        queue for new work. This method is simply pass through and behaves like a
+        submit call as described here `Python docs: <https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor>`_
+
+        Args:
+            - func (callable) : Callable function
+            - *args (list) : List of arbitrary positional arguments.
+
+        Kwargs:
+            - **kwargs (dict) : A dictionary of arbitrary keyword args for func.
+
+        Returns:
+              Future
+        """
+        task_id = uuid.uuid4()
+
+        logger.debug("Pushing function {} to queue with args {}".format(func, args))
+
+        self.tasks[task_id] = Future()
+
+        fn_buf = pack_apply_message(func, args, kwargs,
+                                    buffer_threshold=1024 * 1024,
+                                    item_threshold=1024)
+
+        msg = {"task_id": task_id,
+               "buffer": fn_buf}
+
+        # Post task to the the outgoing queue
+        self.outgoing_q.put(msg)
+
+        # Return the future
+        return self.tasks[task_id]
+
+    @property
+    def scaling_enabled(self):
+        return self._scaling_enabled
+
+    def scale_out(self, workers=1):
+        """Scales out the number of active workers by 1.
+
+        This method is not implemented for threads and will raise the error if called.
+        This would be nice to have, and can be done
+
+        Raises:
+             NotImplementedError
+        """
+        if self.provider:
+            print("[TODO] Scaling out")
+            r = self.provider.submit(self.launch_cmd)
+            self.engines.extend([r])
+        else:
+            logger.error("No execution provider available")
+            r = None
+
+        return r
+
+    def scale_in(self, blocks):
+        """Scale in the number of active blocks by specified amount.
+
+        This method is not implemented for turbine and will raise an error if called.
+
+        Raises:
+             NotImplementedError
+        """
+        to_kill = self.engines[:blocks]
```

Comment:
I know we have an issue open to implement this properly (figure out which workers are idle and kill those). It might be better to just not do anything until we have that worked out, rather than killing engines indiscriminately.